File: Models/CreditCard.cs
using System.Text.Json.Serialization;
using System.Globalization;
using Microcharts;

namespace controle_ja_mobile.Models
{
    public class CreditCard
    {
        [JsonPropertyName("id")]
        public Guid Id { get; set; }

        [JsonPropertyName("accountId")]
        public Guid AccountId { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("totalLimit")]
        public decimal TotalLimit { get; set; }

        [JsonPropertyName("currentLimit")]
        public decimal CurrentLimit { get; set; }

        [JsonPropertyName("closeDay")]
        public int CloseDay { get; set; }

        [JsonPropertyName("bestDay")]
        public int BestDay { get; set; }

        [JsonPropertyName("icon")]
        public string Icon { get; set; } = "credit_card";

        [JsonPropertyName("color")]
        public string Color { get; set; } = "#9C27B0";

        [JsonIgnore]
        public decimal UsedAmount => Math.Max(0, TotalLimit - CurrentLimit);

        [JsonIgnore]
        public string FormattedTotalLimit => $"R$ {TotalLimit.ToString("N2", new CultureInfo("pt-BR"))}";

        [JsonIgnore]
        public string FormattedAvailable => $"R$ {CurrentLimit.ToString("N2", new CultureInfo("pt-BR"))}";

        [JsonIgnore]
        public string FormattedUsed => $"R$ {UsedAmount.ToString("N2", new CultureInfo("pt-BR"))}";

        [JsonIgnore]
        public string InvoiceInfo => $"Fecha dia {CloseDay} • Vence dia {BestDay}";

        [JsonIgnore]
        public double LimitProgress => TotalLimit == 0 ? 0 : (double)UsedAmount / (double)TotalLimit;

        [JsonIgnore]
        public string UsedPercentageText => $"{LimitProgress * 100:F0}% utilizado";

        [JsonIgnore]
        public Chart CategoryChart { get; set; }

        [JsonIgnore]
        public bool HasChartData { get; set; }
    }
}

File: app/src/main/java/com/cainanbt/controleja/data/model/CreditCard.kt
package com.cainanbt.controleja.data.model

import com.google.gson.annotations.SerializedName
import java.text.NumberFormat
import java.util.Locale
import java.util.UUID

data class CreditCard(
    @SerializedName("id") val id: String = UUID.randomUUID().toString(),
    @SerializedName("accountId") val accountId: String = "",
    @SerializedName("name") val name: String = "",
    @SerializedName("totalLimit") val totalLimit: Double = 0.0,
    @SerializedName("currentLimit") val currentLimit: Double = 0.0,
    @SerializedName("closeDay") val closeDay: Int = 1,
    @SerializedName("bestDay") val bestDay: Int = 10,
    @SerializedName("icon") val icon: String = "credit_card",
    @SerializedName("color") val color: String = "#7C4DFF"
) {
    private val formatter = NumberFormat.getCurrencyInstance(Locale("pt", "BR"))

    val usedAmount: Double get() = totalLimit - currentLimit
    val formattedTotalLimit: String get() = formatter.format(totalLimit)
    val formattedAvailable: String get() = formatter.format(currentLimit)
    val formattedUsed: String get() = formatter.format(usedAmount)
    val invoiceInfo: String get() = "Fecha dia $closeDay • Vence dia $bestDay"
    val limitProgress: Float get() = if (totalLimit > 0) (usedAmount / totalLimit).toFloat() else 0f
    val usedPercent: String get() = "${(limitProgress * 100).toInt()}% usado"
}
